### ISSUE TITLE
bot.download_file not downloading

### ISSUE THREAD

**Issue by nathan0** [nickoala/telepot#93]
I have tried to download images, videos or stickers with ```bot.download_file(msg['photo'][-1]['file_id'], './file.png')``` but it seems to get stuck.

**Comment by acidpop** [comment]
i have same issue..

trying bot.download(fileid', 'path)

2 core cpu

CPU usage 51%

**Comment by nathan0** [comment]
I got it to work but had to change __init__.py in teleport/, below are my changes starting at line 470
```
            r = api.download((self._token, f['file_path']))
            d.write(r.data)

            # Ref:
            #   http://stackoverflow.com/questions/17285464/whats-the-best-way-to-download-file-using-urllib3
            #while 1:
            #    data = r.read(self._file_chunk_size)
            #    if data is None:
            #        break
            #    d.write(data)
```

**Comment by nickoala** [comment]
The reason why I did what I did was because I don't want to read all data into memory for a large file. Thanks for looking into the problem, anyway. I have been busy, will start investigating it tomorrow.

**Comment by nickoala** [comment]
Problem solved. The fix will be included in the next version.

For now, please modify two files. First, in `telepot/api.py`, modify the function `download` to allow user-supplied keyword arguments:
```python
def download(req, **user_kw):
    pool = _create_onetime_pool()
    r = pool.request('GET', _fileurl(req), **user_kw)
    return r
```

Second, in `telepot/__init__.py`, modify a section within the method `download_file`:
```python
            r = api.download((self._token, f['file_path']), preload_content=False)

            while 1:
                data = r.read(self._file_chunk_size)
                if not data:
                    break
                d.write(data)
```
Note **two** changes: the `preload_content=False` argument, and `if not data` above the `break`.

You need `preload_content=False` to properly `read()` later on. And `data` is an empty string (not `None`) after stream ends, so we have to check it using `if not data`.

Sorry for the delay. Thanks for using telepot.

**Comment by nickoala** [comment]
Fixed in 8.2
